src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,244 @@
+     * @return a LocalDateTime object with the provided date/time information.
+     * @throws TobiasException if the dateTime string does not follow format of dd-MM-yyyy HHmm.
+     * */
+    public static LocalDateTime dateFromString(String dateTime) throws TobiasException {
+        try {
+            return LocalDateTime.parse(dateTime.trim(), DateTimeFormatter.ofPattern("dd-MM-yyyy HHmm"));
+        } catch (Exception e) {
+            throw new TobiasException("Kindly enter the date or time in this format : dd-MM-yyyy HHmm");
+        }
+    }
+
+    /**
+     * Takes in a command as a String and a TaskList.
+     * Interprets that command and returns the relevant Command object.
+     *
+     * @param command Type of command given as a String
+     * @param tasks TaskList
+     * @throws TobiasException for each possible type of commands and/or if command is not valid.
+     * @return The relevant type of Command object.
+     * */
+    public static Command parseCommands(String command, TaskList tasks) throws TobiasException {
+        if(command.equals("list")) {
+            return new ListCommand();
+        }
+        else if (command.startsWith("todo")) {
+            if (command.equals("todo")) {
+                throw new TobiasException("Hey, please enter a description !");
+            }
+
+            String blank = command.substring(4,5);
+
+            if (!blank.isBlank()) {
+                throw new TobiasException("Kindly type your task a space after todo!");
+            }
+
+            String description = command.substring(5);
+
+            return new TodoCommand(description);
+        }
+        else if (command.startsWith("deadline")) {
+            if (command.equals("deadline")) {
+                throw new TobiasException("Hey, please enter a description !");
+            }
+
+            String blank = command.substring(8,9);
+
+            if (!blank.isBlank()) {
+                throw new TobiasException("Kindly type your task a space after deadline!");
+            }
+
+            int byIndex = command.indexOf("/by");
+
+            if (byIndex == -1) {
+                throw new TobiasException("Hey, please use '/by' to specify the deadline!!");
+            }
+
+            String description = command.substring(9, byIndex);
+
+            int deadlineIndex = byIndex+4;
+
+            if (deadlineIndex > command.length()) {
+                throw new TobiasException("Hey, please enter a date & time in this format : dd-MM-yyyy HHmm !");
+            }
+
+            String deadline = command.substring(byIndex + 4);
+
+            LocalDateTime dd = dateFromString(deadline);
+
+            if (deadline.isEmpty() || deadline.isBlank()) {
+                throw new TobiasException("Hey, please enter a date & time in this format : dd-MM-yyyy HHmm !");
+            }
+
+            return new DeadlineCommand(description, dd);
+        }
+        else if (command.startsWith("event")) {
```

Comment:
Referring to the [Coding Standard](https://se-education.org/guides/conventions/java/intermediate.html) under 

>  The if-else class of statements should have the following form:

```suggestion
        } else if (command.startsWith("event")) {
```